packages/@uppy/utils/tsconfig.json

```
@@ -0,0 +1,10 @@
+{
+  "extends": "../../../tsconfig.shared",
+  "compilerOptions": {
+    "allowJs": true,
+    "emitDeclarationOnly": false,
+    "noEmit": true
+  },
+  "include": ["./package.json", "./types/*.ts"],
```

This doesn't work. Since this is not the build one, we want to check all JS (since we have `allowJS` and otherwise we also have errors in tests) and all TS unconditionally.

```suggestion
  "include": ["**/*.{js,ts}"],
```

---

I think we don't want to `allowJS`, we want to convert JS files to TS

---

You'll probably need it as long as everything isn't migrated yet.

---

Why would we need it? 🤔 IIUC we would miss some errors – but that's fine I think, because those would be errors that are already present. If we add `allowJS`, it forces us to fix the errors, which I think is not a goal, we shouldn't modify the JS files, just convert them to TS when the time has come.

---

All the TS files we write should have exactly 0 untyped JS dependencies, so I really think `allowJS` is not for us.

---

> If we ever import a JS file into a TS that […] would cause the build to fail

Isn't it exactly what we want?

---

For utils now that we are refactoring everything it once, yes. But this comment started about the `include` and I still think that's incorrect.